project/threads/api.py

```diff
@@ -206,44 +207,44 @@ def create_civi(request):
             parent_civi = Civi.objects.get(id=related_civi)
             parent_civi.responses.add(civi)
 
-            if parent_civi.author.user.username != request.user.username:
+            if parent_civi.author.username != request.user.username:
                 notify.send(
                     request.user,  # Actor User
-                    recipient=parent_civi.author.user,  # Target User
+                    recipient=parent_civi.author,  # Target User
                     verb=u"responded to your civi",  # Verb
                     action_object=civi,  # Action Object
                     target=civi.thread,  # Target Object
                     popup_string="{user} responded to your civi in {thread}".format(
-                        user=a.full_name, thread=civi.thread.title
+                        user=user.full_name, thread=civi.thread.title

```

Comment:
I'm having a bit of trouble following the logic here. Can we just use `request.user` her? Is the code behaving as expected?
```suggestion
                        user=request.user.full_name, thread=civi.thread.title

```

Reply:
There were too many lines to change and follow so I intentionally tried not to make huge changes. This is the reason the code is not clear. I'll change all of them with `request.user`.